clients/drcachesim/scheduler/flexible_queue.h

```diff
@@ -96,6 +97,14 @@ class flexible_queue_t {
         return entries_[0]; // Undefined if empty.
     }
 
+    T
+    get_random_entry() // Not const as it change rand_gen's state.
+    {
+        // minstd_rand returns uint_fast32_t.  We do not support get_random_entry()
+        // for queues with >2^32 entries.
+        return entries_[rand_gen_() % size()]; // Undefined if empty.
```

Comment:
I assume we don't want to assert(size() > 0) to keep this path fast?

Reply:
I think that was part of it for top() above too, and that the STL versions do the same thing.  But assert() will go away in release build so I think we can add it.  I added it here and in top() above.